packages/cascara/src/specs/Table/Table.spec.mdx

```diff
@@ -1,7 +1,155 @@
 ---
 title: Table
+status: Draft
+
 ---
 
 # Table
 
-Let's see if this shows up at our application root.
+## Table Types
+
+- Data table The standard table layout with a table header and rows of data
+- Grid view - more of a gallery view. No table header. Could be considered a list. Need to discuss if we want to make this a separate component or a variation on a table.
+
+## Table Locations
+
+### Admin
+
+- BRC (Grid)
+- Storefront Designer (grid)
+- Bundles (grid)
+- Role Bundles (grid)
+- Onboarding Activities/Schedule
+- Onboarding Activities/Performance
+- Service Department
+- Service Teams
+- Barista FAQS
+- Barista FAQS/Topics
+- Users
+- Integrations/CSV/Import History
+- Integrations/CSV/Import Schedule
+- Integrations/Email/Barista Inboxes
+- Reports
+- Locations
+
+### Barista
+
+- None
+
+### Caffeine
+
+- Entities
+- Training
+- Variables
+- Logs (future)
+- Users (future)
+- Configurations (future)
+- Localization(future)
+
+### Status
+
+- Packages
+
+### Doppio
+
+- Applications
+- Applications/[Application]
+- Intents
+- Intents/[Intent]
+- FAQs
+- FAQS/[FAQ]
+- Entities
+- Archetype Conditions
+- Attributes
+- Hooks
+- Synonyms
+- Synonyms/[Synonym]
+- Variables
+- Variables/[Variable]
+- Training
+- Spell Corrector Dictionary
+- Utilities
+- ESP Service Departments
+- Service Team Classifications
+- Task Classifications
+- Service Departments
+- Service Teams
+- Task Categories
+- Testing Reports
+- Test Phrases
+- Test Sets
+- Tasks
+- Tasks/[Task]
+- Categories
+- Categories/[Category]
+- Outages
+- Outages/[Outage]
+- Integrations
+- Integrations/[Integration]
+- Service Requests
+- Service Request/[Service Request]
+- Model Bindings
+- Widgets
+- Users
+- Users/[User]
+- Locations
+- Assets
+- Job Roles
+- Classifiers
+- Data Sources
+- Data Sources/[Data Source]
+- Annotations
+- Annotations/[Annotation]
+- Smart Rules
+- Smart Rules/[Rule]
+- Flagged Tickets
+- Surveys
+- Configurations
+- Attachments
+- Business Events
+- Localization
+
+## Existing Table Functionality
+
+### CRUD Capabilities
+
+- Ability to add a data row
+- Ability to edit a table rown (inline optional)
+- Ability to delete a row (optional)
+
+### Table Header
+
+- Column headers with labels and optional descriptions
+- Sortable column headers
+
+### Pagination
+
+- Ability to show total number of records available
+- Ability to show visible number of records
+- Ability to show how many pages of records are available
+- Ability to navigate to a specific page of results
+- Ability to navigate to next or previous page of results
+
+### Table Row
+
+- Ability to have cells/columns with specific defined data structures, form fields, action elements (buttons, menus)
+- Ability for results view to be accessible via URL so results can be shared
+
+## Needed/Suggested/Future Table Functionality
+
```

Comment:
These are interesting to explore, some of them are already included in the current design, but some others need some more thought. Thanks for adding this section @CaffeinatedAllie, it really opened my mind in terms of architecture. 👍

Reply:
YAYYY!! I think talking big picture/future enhancements right now will really help future proof things. I am glad that what I am doing is helping you!

Reply:
@lordpixel this brings up a good point for us, glad we are calling it out here. Whatever we implement in terms of query params being the state of the filter and pagination... we should make sure that tooling is independent of whatever routing tools we create.

This sounds like a great example of a custom hook. Maybe something like `useQueryState` which composes `useEffect`. This could basically be a value and function. Value would be an object of the current query params as key/values. The function would trigger setting those values based on the object returned. That would make it so we could manage query params 100% independent of router. Then it would work just like useState but it would essentially hoist that state into the url as query params.

There are a number of other areas we will need this logic too.

Reply:
Funny... did a quick search and found this. I do not like that it requires a context provider. I think this could be done as a side effect and be more agnostic to router... but looking into the code, I can see why this does make sense NOT being a side effect and doing this in a more React way. We should bookmark this and come back to it. https://github.com/pbeshai/use-query-params

Reply:
They read your mind @brianespinosa! ok I'll take a look, thanks!